### ISSUE TITLE
embed JEM events on any other site

### ISSUE THREAD

**Issue by Heklaterriol** [jemproject/JEM-Project#1869]
Create the possibility to embed a JEM list of events from a mother site with JEM installed into any other site, without using iframe and regardless of the CMS.
For example with an embedded script, wich is loaded from the mother site

**Comment by Heklaterriol** [comment]
the KI tells me about this:

```
1) Basic Principle
A small JavaScript code is inserted into the target website
This code loads an external JavaScript file from a remote server
The external script contains logic for dynamically rendering the form/content

2) Technical Implementation
<script type="text/javascript" src="https://external-service.com/generated-script.js"></script>

3)Process
The external script identifies a placeholder container on the target page
It dynamically renders the content (e.g., form) within this container
Communicates with the origin server for data and functionality

4) Advantages
No iFrame usage
Dynamic embedding
Easy integration
Central content maintenance
```

**Comment by mckillo** [comment]
An RSS format is available; you just need to add ?format=feed&type=rss to the eventlist view (for example).

![Image](https://github.com/user-attachments/assets/f5ae9473-35bd-4683-80f1-f9b4cf4c45b6)

This allows you to import the event to another website via RSS (XML). However, JEM currently doesn't include an import class for this. It could be a useful feature for a future JEM update. An idea.

**Comment by Heklaterriol** [comment]
I do some experiements and testing. I let you know as soon as I have some results.

**Comment by mckillo** [comment]
I think that a possible solution is "Adding RSS Feed Support to Joomla Events Manager (JEM) Modules"
**Suggestion**: Modify any JEM (Joomla Events Manager) module to include an RSS URL input field, allowing the module to fetch and display events directly from an external RSS feed.
**Alternatives**: Develop a custom JEM plugin (instead of modifying core modules).

**Benefits**:
- Dynamic Event Integration – Pull events from external sources (e.g., Meetup, Eventbrite, or custom calendars) without manual entry.
- Reduced Manual Work – Automatically sync events instead of duplicating them in JEM.
- Flexible Content Aggregation – Display events from multiple sources in a single JEM module.

**Comment by Heklaterriol** [comment]
I created a plugin to create a JSON output with the parameters in the URL and a security token, e.g.
https://domain.com/index.php?option=com_ajax&plugin=jemembed&group=content&format=json&token=ABCDEFG&type=upcoming&max=10&catids=1,2,3&title=link&date=on&venue=link&category=link&cuttitle=30

Then on the target site I suggest a  a proxy script `feed-proxy.php` to load the date and avoid CORS issues (Cross-Origin Resource Sharing) and a JavaScript file to prepare the output. This way the events it can be embeded into a html page with a simple code snippet.

**Comment by jojo12** [comment]
question: can this be a possibility to embed JEM events in Acymailing (you can add there Joomla articles)?

**Comment by Heklaterriol** [comment]
> question: can this be a possibility to embed JEM events in Acymailing (you can add there Joomla articles)?

That is a different approach. But a s fa as I can see, Acymailing provides information how to create such a plugin.

**Comment by jojo12** [comment]
Yes, but e.g. I've seen even wzthout a plugin, you can import a module, even a JEM module, which is shown then in the newsletter!

**Comment by Heklaterriol** [comment]
To keep you up to date:

The JEM Embed plugin is already well advanced, but I'm not quite satisfied with the handling yet

So far it works as follows:
The plugin outputs JEM Events as JSON. These output data can be used in various ways. My interest, and what I also need, is that I can embed the events in any external page:

Default JSON output parameters are defined in the plugin configuration, simlar to the ones of jemlistevents. 
In addition, a security token to prevent unallowed queries must be defined in the parameters.
The JSON output can be individually modified by adding and adjusting the parameters in the output URL.

**Example:**
The base URL is:
https://domain.com/index.php?option=com_ajax&plugin=jemembed&group=content&format=json&token=SECURITY_TOKEN

with modifying params, e.g.
https://domain.com/index.php?option=com_ajax&plugin=jemembed&group=content&format=json&token=SECURITY_TOKEN&max=10&venueids=3,1&category=2,5

**Possible parameters in the URL** 
type=today|unfinished|upcoming|archived|newest
featured=on|off|only
title=on|link|off
date=on|link|off
time=on|off
enddatetime=on|off
catids=[comma separated category ids]
category=on|link|off
venueids=[comma separated venue ids]
venue=on|link|off
max=n
cuttitle=n
noeventsmsg=[message text]

The parameters are combined with AND, i.e. only those events are displayed that fulfill **all** conditions.


**How to embed the JEM events in external website:**
1. downloaded jemevents.zip from `https://www.sourcedomain.com/media/plg_content_jemembed/files/jemevents.zip` and unzip it and you get three files: jemevents.php, jemevents.js and jemevents-example.html
3. Upload jemevents.php and jemevents.js to the web space of the external website. The file jemevents-example.html is just an example page with embedded events.
4. Fill in the settings part in jemevents.php
5. In the page, where the events are to embed, insert the link to the jemevents.js file before the `</body>` end tag. e.g. `<script src="/jemevents.js"></script>`
6. In the place, where the events are to be embedded insert  `<div id="jemevents-container"></div>`

**Comment by jojo12** [comment]
JEMembedJSON_4.3.3 postponed to 4.4.0
